### PR TITLE
Update EKAT submodule

### PR DESCRIPTION
**Pull request by jgfouca**
to eeaac1a32de15c25b0fdb895b9872ff31702ff7b

Changes:
1) Update YAML-cpp submodule to remove non-portable warning flags
2) Fix GetMpiDistributionName for Cray mpi compilers
3) Change default constructor of Pack to init to 0
4) Use check_symbol_exists instead of try_compile to detect MPI distro name
5) Fix GetMpiDistributionName
6) Allow forcing build of tpls from internal submodules
7) Add native support for WILL_FAIL in EkatCreateUnitTest
8) Ensure that yaml is always linked using the scoped target name
9) Do not use new find_package option with old cmake
10) [SYCL] Minor cleanup for IntelLLVM cmake
11) Some fixes that allow to build with raw compilers instead of mpi wrap…
12) Error out if the found kokkos installation is not compatible with the requested options
13) Add missing quotes for FIXTURES to cmake function create unit test.

To sum up, it looks like nearly all the changes are build/cmake related. This is probably safe to merge IMO.

### CODE REVIEW

**Review by bartgol — APPROVED** [review]
I agree, these should be safe to merge. If, by chance, we break the build on some machine, we should be able to ninja fix it. It's not like causing hard-to-find non bfb diffs...

**Comment by bartgol** [comment]
Ah, we need to update the tests that should fail, since WILL_FAIL is now an ecut option.

**Comment by jgfouca** [comment]
@bartgol , fixed. Also fixed an issue with cprnc injecting its ctests into ours.

**Comment by bartgol** [comment]
> @bartgol , fixed. Also fixed an issue with cprnc injecting its ctests into ours.

Sorry, I just saw this. Do you want to still do E3SM-Project/ekat#321? Or should I take back the change?

**Comment by jgfouca** [comment]
@bartgol , I actually like `WILL_FAIL` better than `WILL_FAIL TRUE`. I would prefer you close that PR.

**Comment by bartgol** [comment]
> @bartgol , I actually like `WILL_FAIL` better than `WILL_FAIL TRUE`. I would prefer you close that PR.

Ok. I will change it back. But that PR has another fix, which would be nice to incorporate. I will close and reopen just with the fix.

**Review by bartgol — APPROVED** [review]
